src/main/java/com/iexec/common/security/CipherUtils.java

```diff
@@ -200,48 +204,47 @@ public static byte[] aesEncrypt(
      * <p>
      * Note: PKCS5Padding is equivalent to PKCS7Padding
      * <p>
-     * Ref: https://stackoverflow.com/a/10194082/7631879
+     * Ref: <a href="https://stackoverflow.com/a/10194082/7631879">https://stackoverflow.com/a/10194082/7631879</a>
      *
      * @param ivAndEncryptedData to decrypt
-     * @param base64Key          Base64 encoded AES key
+     * @param key                AES key
      * @return Decrypted data
-     * @throws GeneralSecurityException
+     * @throws GeneralSecurityException if an error occurs during AES decryption.
      * @throws IllegalArgumentException if the data's length
      *                                  is less than 16 bytes.
      * @see <a href="https://stackoverflow.com/a/34004582">AES encryption on file over 1GB</a> for large files
      */
-    public static byte[] aesDecrypt(byte[] ivAndEncryptedData, byte[] base64Key)
+    public static byte[] aesDecrypt(byte[] ivAndEncryptedData, byte[] key)
             throws GeneralSecurityException {
-        Objects.requireNonNull(ivAndEncryptedData, "Data cannot be null");
-        Objects.requireNonNull(base64Key, "Base64 AES key cannot be null");
+        Objects.requireNonNull(ivAndEncryptedData, NOT_NULL_DATA);
+        Objects.requireNonNull(key, "Base64 AES key cannot be null");
         if (ivAndEncryptedData.length < 16) {
             throw new IllegalArgumentException("Data cannot be less than 16 bytes");
         }
-        byte[] decodedKey = Base64.getDecoder().decode(base64Key);
         byte[] iv = getIvFromEncryptedData(ivAndEncryptedData);
         byte[] data = stripIvFromEncryptedData(ivAndEncryptedData);
-        return aesDecrypt(data, decodedKey, iv);
+        return aesDecrypt(data, key, iv);
     }
 
     /**
      * Decrypt data with AES256/CBC/PKCS5Padding.
      * <p>
      * Note: PKCS5Padding is equivalent to PKCS7Padding
      * <p>
-     * Ref: https://stackoverflow.com/a/10194082/7631879
+     * Ref: <a href="https://stackoverflow.com/a/10194082/7631879">https://stackoverflow.com/a/10194082/7631879</a>
      *
-     * @param plainData
-     * @param key
-     * @param iv
-     * @return
-     * @throws GeneralSecurityException
+     * @param plainData Data to decrypt
+     * @param key       AES key
+     * @param iv        Initialization vector
+     * @return Decrypted data
+     * @throws GeneralSecurityException if a security related error occurs.
      */
     public static byte[] aesDecrypt(
```

Comment:
I think this method could become `private`. I found no usage of this method except from the method above.
This will allow to diminish redundant checks on `null`.
```suggestion
    private static byte[] aesDecrypt(
```
This is another way to deal with the previous issues on duplicated strings.

Reply:
Resolved in [cc09d9b](https://github.com/iExecBlockchainComputing/iexec-common/pull/470/commits/cc09d9b58dd26d71e3c901ac02505ef4264cb657)

Reply:
Resolved in [cc09d9b](https://github.com/iExecBlockchainComputing/iexec-common/pull/470/commits/cc09d9b58dd26d71e3c901ac02505ef4264cb657)